src/compile/selection/selection.ts

```diff
@@ -211,13 +214,23 @@ const PREDICATES_OPS = {
   intersect_others: '"intersect", "others"'
 };
 
-export function predicate(model: Model, selections: LogicalOperand<string>): string {
+export function predicate(model: Model, selections: LogicalOperand<string>, dfnode?: DataFlowNode): string {
   function expr(name: string): string {
     const vname = varName(name);
     const selCmpt = model.getSelectionComponent(vname, name);
     const store = stringValue(vname + STORE);
     const op = PREDICATES_OPS[selCmpt.resolve];
 
+    if (selCmpt.timeUnit) {
```

Comment:
We just observed that this is basically making the `predicate` method no longer pure and thus cause side-effect to the dataflow.    If possible, we should decouple this.  I'll file an issue.